### DISCUSSION TITLE
Isolate server

### DISCUSSION THREAD

**Discussion by jolly-g** [sensmetry/sysml-2ls#9] [Q&A]
Hello,
I'm looking for a way to launch the SysML language server without the VSCode extension. I would like to implement my own client in Java which can communicate with the language server. 
Is there currently a way to start it alone or not ?

**Comment by daumantas-kavolis-sensmetry** [comment]
See https://github.com/sensmetry/sysml-2ls/tree/main/packages/syside-languageserver/src/node, VS Code simply launches the language server through `node` wrapper with appropriate CLI arguments. Script `esbuild:node` in `syside-vscode` is used to build the language server for VS Code extension and it produces a standalone (excluding `node` runtime) language server in `dist/node/language-server`.

**Reply by GuillaumeEscande** [reply]
Hi, I am trying this method to run the language server on a local socket. It seems that the language server socket never goes up, and the language server stops before creating the socket.

While using the debugger, the problem seems to originate here:
[Link to the specific code on GitHub] (https://github.com/eclipse-langium/langium/blob/main/packages/langium/src/lsp/language-server.ts#L232)
The language server stops running when this instruction is executed.

Do you have any ideas on how to correctly use the standalone language server?

Thanks.

**Reply by daumantas-kavolis-sensmetry** [reply]
Standalone server can be downloaded from CI pipelines on GitLab, e.g. https://gitlab.com/sensmetry/public/sysml-2ls/-/jobs/5496527102/artifacts/file/syside-languageserver.js. The command line should look like `node syside-languageserver.js --socket SOCKET [--hostname HOSTNAME] [--clientProcessId PID]`, you may also try communicating through a named pipe with `--pipe PIPE` instead of socket. I don't known how much I can help other than that.

**Reply by GuillaumeEscande** [reply]
Thanks for the reply.

When I try this script, I have the same behaviour, the socket seams to be close instantly.
Does this script must be run somewhere special ? Or with specific folder / files aside ? Does the client must be run before to start the server ?

Thanks again !

**Reply by daumantas-kavolis-sensmetry** [reply]
I'm not sure how Node net module works and why it closes the connection instantly, perhaps it exists because there was no client. As a workaround, `--stdio` doesn't exist immediately.
